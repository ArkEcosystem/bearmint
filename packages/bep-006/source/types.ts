import type {
	Account,
	AccountSerializer,
	AddressFactory,
	Container,
	EventDispatcher,
	EventRecorder,
	KeyPair,
	ServiceProviderRepository,
	StakingService,
	StateStore,
	Tx,
	TxReceipt,
	VoidPromiseFn,
} from '@bearmint/bep-013'

export interface TxTestContext {
	accountSerializer: AccountSerializer
	accountFromMnemonic: (mnemonic: string) => Promise<{ account: Account; keypair: KeyPair }>
	addressFactory: AddressFactory
	audit: (tx: Tx) => Promise<VoidPromiseFn>
	client: any
	container: Container
	eventDispatcher: EventDispatcher
	eventRecorder: EventRecorder
	execute: (tx: Tx) => Promise<VoidPromiseFn>
	executeFull: (tx: Tx) => Promise<void>
	makeTx: (
		sender: KeyPair,
		recipient: KeyPair,
		content: Uint8Array,
		handler?: string | undefined,
		version?: string | undefined,
	) => Promise<Tx>
	receipt: TxReceipt
	serviceProviderRepository: ServiceProviderRepository
	stakingService: StakingService
	state: StateStore
}
