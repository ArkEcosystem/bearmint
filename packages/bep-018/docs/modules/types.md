[@bearmint/bep-018 - v0.0.0](../README.md) / types

# Namespace: types

## Table of contents

### Enumerations

- [BlockIDFlag](../enums/types.BlockIDFlag.md)
- [SignedMsgType](../enums/types.SignedMsgType.md)

### Classes

- [BlockID](../classes/types.BlockID.md)
- [BlockMeta](../classes/types.BlockMeta.md)
- [Commit](../classes/types.Commit.md)
- [CommitSig](../classes/types.CommitSig.md)
- [Data](../classes/types.Data.md)
- [Header](../classes/types.Header.md)
- [LightBlock](../classes/types.LightBlock.md)
- [Part](../classes/types.Part.md)
- [PartSetHeader](../classes/types.PartSetHeader.md)
- [Proposal](../classes/types.Proposal.md)
- [SignedHeader](../classes/types.SignedHeader.md)
- [TxProof](../classes/types.TxProof.md)
- [Vote](../classes/types.Vote.md)
