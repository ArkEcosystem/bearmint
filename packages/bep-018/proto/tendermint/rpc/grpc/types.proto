syntax = "proto3";
package tendermint.rpc.grpc;

import "tendermint/abci/types.proto";

option go_package = "github.com/tendermint/tendermint/rpc/grpc;coregrpc";

//----------------------------------------
// Request types

message RequestPing {}

message RequestBroadcastTx {
  bytes tx = 1;
}

//----------------------------------------
// Response types

message ResponsePing {}

message ResponseBroadcastTx {
  tendermint.abci.ResponseCheckTx check_tx = 1;
  tendermint.abci.ExecTxResult tx_result = 2;
}

//----------------------------------------
// Service Definition

service BroadcastAPI {
  rpc Ping(RequestPing) returns (ResponsePing);
  rpc BroadcastTx(RequestBroadcastTx) returns (ResponseBroadcastTx);
}
