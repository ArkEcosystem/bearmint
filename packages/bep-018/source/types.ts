/**
 * @see https://github.com/bufbuild/protobuf-es/issues/230
 */

import type { Message } from '@bufbuild/protobuf'

// prettier-ignore
type PlainField<F> =
  F extends (Date | Uint8Array | bigint | boolean | string | number) ? F
  : F extends Array<infer U> ? Array<PlainField<U>>
  : F extends ReadonlyArray<infer U> ? ReadonlyArray<PlainField<U>>
  : F extends Message ? Plain<F>
  : F extends OneofSelectedMessage<infer C, infer V> ? { case: C; value: Plain<V> }
  : F extends { case: string | undefined; value?: unknown } ? F
  : F extends { [key: string|number]: Message<infer U> } ? { [key: string|number]: Plain<U> }
  : F;

interface OneofSelectedMessage<K extends string, M extends Message> {
	case: K
	value: M
}

export type Plain<T extends Message> = {
	// eslint-disable-next-line @typescript-eslint/ban-types
	[P in keyof T as T[P] extends Function ? never : P]: PlainField<T[P]>
}
