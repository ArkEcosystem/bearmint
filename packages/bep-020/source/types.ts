// Last Block (Mutated during TM_COMMIT)
export const TK_COMMITTED_BLOCK = '@bearmint/bep-089/committed_block'

// Milestone (Mutated during TM_END_BLOCK)
export const TK_MILESTONE = '@bearmint/bep-021/milestone'
