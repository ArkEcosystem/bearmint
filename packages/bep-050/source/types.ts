export interface CostParameters {
	msg: string
	tx: string
}

export interface Milestone {
	[key: string]: Record<string, CostParameters>
}
