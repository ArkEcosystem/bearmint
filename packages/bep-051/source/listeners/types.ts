import type { EventRecorder, GasMeter, StateStore, Tx } from '@bearmint/bep-013'

export interface TxListenerPayload<T = unknown> {
	eventRecorder: EventRecorder
	gasMeter: GasMeter
	msg: T
	state: StateStore
	tx: Tx
}
