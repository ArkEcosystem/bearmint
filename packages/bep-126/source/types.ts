import { readPackageJson } from '@bearmint/bep-009'
import type { Account, StateStore } from '@bearmint/bep-013'
import type { BEP126Collection, BEP126Policy } from '@bearmint/bep-018'

export const HANDLER = '@bearmint/bep-126'
export const VERSION = '0.0.0'

export const MODULE_VERSION = readPackageJson(import.meta.url).version

export interface PolicyArgs {
	actor: Account
	collection: BEP126Collection
	policy: BEP126Policy
	state: StateStore
	token?: string
}

export enum MintPolicy {
	IMMUTABLE = '@bearmint/bep-142',
	LOCKED_BY_BLOCK_NUMBER = '@bearmint/bep-143',
	LOCKED_BY_MAX_SUPPLY = '@bearmint/bep-144',
	ONLY_OWNER = '@bearmint/bep-141',
	ONLY_TEAM = '@bearmint/bep-145',
}

export enum BEP126UserRole {
	OWNER,
	STEWARD,
}

export enum BEP126UserPermission {
	BURN,
	MINT,
	TRANSFER,
}
