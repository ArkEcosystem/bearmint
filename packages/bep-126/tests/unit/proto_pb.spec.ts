import { expect, describe, it } from 'vitest'
import { extendExpect } from '@bearmint/bep-005'
extendExpect(expect)
import { BEP126Collection, BEP131Token, BEP126User } from '@bearmint/bep-018'

import { BEP126UserPermission, BEP126UserRole } from '../../source/types.js'
import { STUB_TX_META } from '../fixtures.js'

describe('Collection', () => {
	it('should serialize and deserialize', () => {
		const actual = BEP126Collection.fromBinary(new BEP126Collection(STUB_TX_META).toBinary())

		expect(actual.toBinary()).toMatchSnapshot()
		expect(actual.toJson()).toMatchSnapshot()
		expect(actual.toJsonString()).toMatchSnapshot()
	})
})

describe('BEP131Token', () => {
	it('should serialize and deserialize', () => {
		const actual = BEP131Token.fromBinary(
			new BEP131Token({
				collection: 'collection',
				id: 1,
				owner: 'owner',
			}).toBinary(),
		)

		expect(actual.toBinary()).toMatchSnapshot()
		expect(actual.toJson()).toMatchSnapshot()
		expect(actual.toJsonString()).toMatchSnapshot()
	})
})

describe('User', () => {
	it('should serialize and deserialize', () => {
		const actual = BEP126User.fromBinary(
			new BEP126User({
				permissions: [
					BEP126UserPermission.BURN,
					BEP126UserPermission.MINT,
					BEP126UserPermission.TRANSFER,
				],
				role: BEP126UserRole.OWNER,
			}).toBinary(),
		)

		expect(actual.toBinary()).toMatchSnapshot()
		expect(actual.toJson()).toMatchSnapshot()
		expect(actual.toJsonString()).toMatchSnapshot()
	})
})
