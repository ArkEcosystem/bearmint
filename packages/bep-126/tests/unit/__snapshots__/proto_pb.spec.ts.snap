// Vitest Snapshot v1

exports[`BEP131Token > should serialize and deserialize 1`] = `
Uint8Array [
  8,
  1,
  26,
  10,
  99,
  111,
  108,
  108,
  101,
  99,
  116,
  105,
  111,
  110,
  34,
  5,
  111,
  119,
  110,
  101,
  114,
]
`;

exports[`BEP131Token > should serialize and deserialize 2`] = `
{
  "collection": "collection",
  "id": 1,
  "owner": "owner",
}
`;

exports[`BEP131Token > should serialize and deserialize 3`] = `"{\\"id\\":1,\\"collection\\":\\"collection\\",\\"owner\\":\\"owner\\"}"`;

exports[`Collection > should serialize and deserialize 1`] = `
Uint8Array [
  10,
  64,
  101,
  50,
  97,
  98,
  101,
  56,
  51,
  49,
  52,
  57,
  52,
  98,
  100,
  54,
  100,
  56,
  54,
  52,
  50,
  56,
  49,
  48,
  99,
  98,
  49,
  49,
  101,
  50,
  100,
  51,
  54,
  57,
  52,
  52,
  97,
  98,
  98,
  50,
  50,
  48,
  54,
  99,
  52,
  48,
  97,
  99,
  53,
  102,
  50,
  52,
  98,
  101,
  55,
  51,
  100,
  57,
  50,
  50,
  53,
  56,
  100,
  102,
  97,
  57,
  18,
  4,
  110,
  97,
  109,
  101,
  26,
  6,
  115,
  121,
  109,
  98,
  111,
  108,
  34,
  9,
  117,
  114,
  105,
  80,
  114,
  101,
  102,
  105,
  120,
  42,
  9,
  117,
  114,
  105,
  83,
  117,
  102,
  102,
  105,
  120,
  58,
  88,
  98,
  101,
  97,
  114,
  49,
  53,
  122,
  50,
  48,
  99,
  119,
  99,
  103,
  115,
  122,
  114,
  53,
  52,
  118,
  55,
  119,
  55,
  112,
  120,
  117,
  108,
  121,
  118,
  51,
  112,
  121,
  113,
  106,
  122,
  56,
  55,
  56,
  114,
  109,
  118,
  122,
  113,
  116,
  106,
  102,
  50,
  115,
  113,
  112,
  104,
  122,
  55,
  103,
  118,
  52,
  104,
  114,
  122,
  48,
  102,
  52,
  52,
  118,
  117,
  119,
  103,
  120,
  53,
  55,
  113,
  121,
  54,
  104,
  48,
  99,
  48,
  54,
  52,
  120,
  51,
  114,
  119,
  52,
  52,
  122,
  120,
  112,
  108,
  66,
  88,
  98,
  101,
  97,
  114,
  49,
  53,
  122,
  50,
  48,
  99,
  119,
  99,
  103,
  115,
  122,
  114,
  53,
  52,
  118,
  55,
  119,
  55,
  112,
  120,
  117,
  108,
  121,
  118,
  51,
  112,
  121,
  113,
  106,
  122,
  56,
  55,
  56,
  114,
  109,
  118,
  122,
  113,
  116,
  106,
  102,
  50,
  115,
  113,
  112,
  104,
  122,
  55,
  103,
  118,
  52,
  104,
  114,
  122,
  48,
  102,
  52,
  52,
  118,
  117,
  119,
  103,
  120,
  53,
  55,
  113,
  121,
  54,
  104,
  48,
  99,
  48,
  54,
  52,
  120,
  51,
  114,
  119,
  52,
  52,
  122,
  120,
  112,
  108,
  74,
  97,
  10,
  88,
  98,
  101,
  97,
  114,
  49,
  53,
  122,
  50,
  48,
  99,
  119,
  99,
  103,
  115,
  122,
  114,
  53,
  52,
  118,
  55,
  119,
  55,
  112,
  120,
  117,
  108,
  121,
  118,
  51,
  112,
  121,
  113,
  106,
  122,
  56,
  55,
  56,
  114,
  109,
  118,
  122,
  113,
  116,
  106,
  102,
  50,
  115,
  113,
  112,
  104,
  122,
  55,
  103,
  118,
  52,
  104,
  114,
  122,
  48,
  102,
  52,
  52,
  118,
  117,
  119,
  103,
  120,
  53,
  55,
  113,
  121,
  54,
  104,
  48,
  99,
  48,
  54,
  52,
  120,
  51,
  114,
  119,
  52,
  52,
  122,
  120,
  112,
  108,
  18,
  5,
  18,
  3,
  0,
  1,
  2,
]
`;

exports[`Collection > should serialize and deserialize 2`] = `
{
  "accessControlList": {
    "bear15z20cwcgszr54v7w7pxulyv3pyqjz878rmvzqtjf2sqphz7gv4hrz0f44vuwgx57qy6h0c064x3rw44zxpl": {
      "permissions": [
        0,
        1,
        2,
      ],
    },
  },
  "currentOwner": "bear15z20cwcgszr54v7w7pxulyv3pyqjz878rmvzqtjf2sqphz7gv4hrz0f44vuwgx57qy6h0c064x3rw44zxpl",
  "id": "e2abe831494bd6d8642810cb11e2d36944abb2206c40ac5f24be73d92258dfa9",
  "initialOwner": "bear15z20cwcgszr54v7w7pxulyv3pyqjz878rmvzqtjf2sqphz7gv4hrz0f44vuwgx57qy6h0c064x3rw44zxpl",
  "name": "name",
  "symbol": "symbol",
  "uriPrefix": "uriPrefix",
  "uriSuffix": "uriSuffix",
}
`;

exports[`Collection > should serialize and deserialize 3`] = `"{\\"id\\":\\"e2abe831494bd6d8642810cb11e2d36944abb2206c40ac5f24be73d92258dfa9\\",\\"name\\":\\"name\\",\\"symbol\\":\\"symbol\\",\\"uriPrefix\\":\\"uriPrefix\\",\\"uriSuffix\\":\\"uriSuffix\\",\\"initialOwner\\":\\"bear15z20cwcgszr54v7w7pxulyv3pyqjz878rmvzqtjf2sqphz7gv4hrz0f44vuwgx57qy6h0c064x3rw44zxpl\\",\\"currentOwner\\":\\"bear15z20cwcgszr54v7w7pxulyv3pyqjz878rmvzqtjf2sqphz7gv4hrz0f44vuwgx57qy6h0c064x3rw44zxpl\\",\\"accessControlList\\":{\\"bear15z20cwcgszr54v7w7pxulyv3pyqjz878rmvzqtjf2sqphz7gv4hrz0f44vuwgx57qy6h0c064x3rw44zxpl\\":{\\"permissions\\":[0,1,2]}}}"`;

exports[`User > should serialize and deserialize 1`] = `
Uint8Array [
  18,
  3,
  0,
  1,
  2,
]
`;

exports[`User > should serialize and deserialize 2`] = `
{
  "permissions": [
    0,
    1,
    2,
  ],
}
`;

exports[`User > should serialize and deserialize 3`] = `"{\\"permissions\\":[0,1,2]}"`;
